### ISSUE TITLE
 best-effort flush events at shutdown

### ISSUE THREAD

**Issue by yufengwangca** [project-chip/connectedhomeip#13008]
#### Problem
Currently, we don't flush event during shut down, and all pending events will  get lost after device is shut down.

 #### Proposed Solution
 flush out the event buffer on shutdown

**Comment by bzbarsky-apple** [comment]
In particular, we should only flush as much of the event buffer as will fit in a single packet.  No chunking, etc.  Just do what we can.

**Comment by andy31415** [comment]
Marked as V1 since I understand that without this, CSG and test tooling would have a harder time testing events.
If this is not the case or there are workarounds, lets move it back to 1.x

**Comment by kkasperczyk-no** [comment]
@yufengwangca are you sure this https://github.com/project-chip/connectedhomeip/pull/15331 solves the issue? Sorry if I'm wrong, but I tried to log some event and flush it on shutdown and it seems that it doesn't work. What is not clear for me how it should work if `ScheduleUrgentEventDeliverySync` is called in `Engine::Shutdown` and after `InteractionModelEngine` shutdown, so all handlers are deinitialized? I did some hacks and called `ScheduleUrgentEventDeliverySync` from my application without shutdown and in that case it indeed works great, but I don't understand how it should work when called after the shutdown.
